flink-runtime/src/main/scala/org/apache/flink/runtime/jobmanager/JobManager.scala

```diff
@@ -536,6 +536,11 @@ class JobManager(
         )
       )
 
+    case RequestTaskManagerInstance(instanceID) =>
+      sender ! decorateMessage(
+        TaskManagerInstance(instanceManager.getRegisteredInstanceById(instanceID))
+      )
+
```

Comment:
Wouldn't it be better to return an InstanceNotFound message instead of a null instance?